LuaMenu/widgets/chobby/components/battle/battle_list_window.lua

```diff
@@ -661,17 +661,19 @@ function BattleListWindow:FilterRedundantBattle(battle, id)
 	local lowestEmptyBattleID = nil
 	for k, otherBattleID in pairs(self.allBattleIDs) do
 		local otherBattle = lobby:GetBattle(otherBattleID)
-		local ob_hostCountry, ob_hostnumber = parseBattleNumber(otherBattle.title)
-		local otherBattlePlayerCount = lobby:GetBattlePlayerCount(otherBattleID)
-
-		if ob_hostCountry and
-			ob_hostnumber < lowestEmptyBattleIndex and
-			otherBattlePlayerCount == 0 and
-			otherBattle.spectatorCount == 1 and
-			myCountry == ob_hostCountry then
-
-			lowestEmptyBattleID = otherBattleID
-			lowestEmptyBattleIndex = ob_hostnumber
+		if otherBattle then
```

Comment:
I don't see how this can become nil unless there's a bug somewhere else?

From what I can tell this is only invoked from SoftUpdate:

```
function BattleListWindow:SoftUpdate()
	if Configuration.battleFilterRedundant then
		self:UpdateAllBattleIDs() -- this updates the allBattleIDs
	end
	self:UpdateFilters() -- this uses allBattleIDs by calling ItemInFilter which calls FilterRedundantBattle
	self:UpdateInfoPanel()
end
```

I don't see how the battle can become invalid within the same function call

Reply:
The nil check was added because it became nil, for some reason, and I wanted to ensure that the battles werent hidden because of possible bugs in my filtering code.

Reply:
can you print something when it's nil? it's a likely bug and I'd like to figure out why and when it's happening

Reply:
Sure